### PR TITLE
[UI] Improved Statistics Scenarios

### PR DESCRIPTION
**Pull request by Rafikooo**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | 1.13                   |
| Bug fix?        | no                                                      |
| New feature?    | no                                                      |
| BC breaks?      | no                                                      |
| Deprecations?   | no |
| Related tickets | https://github.com/Sylius/Sylius/pull/15598   |
| License         | MIT                                                          |

Viewing statistics across various time periods was not being tested. This pull request also aims to simplify the testing process for the API feature counterpart that is introduced here: https://github.com/Sylius/Sylius/pull/15598

### CODE REVIEW

File: src/Sylius/Behat/Page/Admin/DashboardPage.php [hunk 1]
```diff
@@ -13,91 +13,141 @@
...
+    {
+        $this->getElement('month_split_by_days_statistics_button')->click();
+    }
+
+    /** @throws ElementNotFoundException */
+    public function choosePreviousPeriod(): void
+    {
+        $this->getElement('navigation_previous')->click();
+
+        usleep(500000);
```
Comment (Wojdylak) [inline src/Sylius/Behat/Page/Admin/DashboardPage.php:116]: Why 500000? Could we possibly adopt an approach similar to what is used in `JavaScriptTestHelper`?

Comment (Rafikooo) [reply src/Sylius/Behat/Page/Admin/DashboardPage.php:116]: I'm not certain, as the element is consistently present when selenium (I hope it's selenium) clicks it, and it retrieves the value properly, but not the correct one, so I opted to insert this slight delay

File: src/Sylius/Behat/Context/Ui/Admin/DashboardContext.php [hunk 2]
```diff
@@ -38,17 +39,50 @@ public function iOpenAdministrationDashboard(): void
...
-     * @When I open administration dashboard for :name channel
+     * @When I view statistics for :name channel
+     *
+     * @throws UnexpectedPageException
      */
-    public function iOpenAdministrationDashboardForChannel($name)
+    public function iViewStatisticsForChannel(string $channelName): void
     {
-        $this->dashboardPage->open(['channel' => StringInflector::nameToLowercaseCode($name)]);
+        $channelName === 'United States' ?
```
Comment (jakubtobiasz) [inline src/Sylius/Behat/Context/Ui/Admin/DashboardContext.php]: You can use `\Sylius\Behat\Context\Transform\ChannelContext` transformer here.

File: src/Sylius/Behat/Context/Ui/Admin/DashboardContext.php [hunk 3]
```diff
@@ -38,17 +39,50 @@ public function iOpenAdministrationDashboard(): void
...
+     * @When /^I view statistics for "([^"]+)" channel and (current|previous|next) year split by (month|day)$/
+     *
+     * @throws UnexpectedPageException
+     */
+    public function iViewStatisticsForChannelAndYear(
+        string $channelName,
+        string $period,
+        string $interval,
+    ): void {
+        $channelName === 'United States'
```
Comment (jakubtobiasz) [inline src/Sylius/Behat/Context/Ui/Admin/DashboardContext.php]: As above, apply to all cases.

File: src/Sylius/Behat/Page/Admin/DashboardPage.php (hunk 1 above)
Comment (jakubtobiasz) [reply src/Sylius/Behat/Page/Admin/DashboardPage.php:116]: It isn't Selenium. We're using chromedriver.

Comment (jakubtobiasz) [reply src/Sylius/Behat/Page/Admin/DashboardPage.php:116]: Anyway, I have mixed feeling about it, but no better idea for now.

Comment (NoResponseMate) [reply src/Sylius/Behat/Page/Admin/DashboardPage.php:116]: You could try using `JQueryHelper::waitForAsynchronousActionsToFinish` or wait until the `active` class disappears from the `stats-loader` element; and as a side note, why usleep? It looks so weird having half a million in code

**Review by jakubtobiasz — APPROVED** [review]

**Review by NoResponseMate — APPROVED** [review]

**Comment by jakubtobiasz** [comment]
Thank you, @Rafikooo!

**Pull request merged**